src/refda/exec.c

```diff
@@ -375,91 +376,260 @@ void *refda_exec_worker(void *arg)
     // run until explicitly told to stop via refda_agent_t::execs
     while (true)
     {
-        refda_msgdata_t item;
+        refda_exec_worker_iteration(agent);
+    }
 
-        refda_timeline_it_t tl_it;
-        refda_timeline_it(tl_it, agent->exec_timeline);
-        if (!refda_timeline_end_p(tl_it))
+    CACE_LOG_INFO("Worker stopped");
+    return NULL;
+}
+
+void refda_exec_worker_iteration(refda_agent_t *agent)
```

Comment:
Good refactoring, but may need to have some return status to know when to exit the thread loop (see earlier comment about thread completion). Some other APIs return `true` to continue looping and `false` when signaled to stop.

Reply:
Thanks, refactored to take this into account.